contribute/design/1.x/csi/20190606-csi-volume-provisioning.md

```diff
@@ -254,134 +247,141 @@ Below represents resources that are in a constant state of reconciliation
-spec:
-  targetDeployment:
-    spec:
-      containers:
-      - name: cstor-istgt
-        env:
-        - name: QueueDepth
-          value: 6
-        - name: Luworkers
-          value: 3
-  targetService:
-    spec:
-      labels:
-      annotations:
-  cv:
-    spec:
-      labels:
-      annotations:
-  cvr:
-    spec:
-      labels:
-      annotations:
 ```
 
 ##### NOTES
-- `spec.targetDeployment.spec`, `spec.replica.spec` and `spec.targetService.spec` 
-will be of same data type
-- It will have below fields:
-  - labels,
-  - annotations,
-  - env,
-  - owners,
-  - nodeAffinity,
-  - podAntiAffinity,
-  - containers,
-  - initContainers,
-  - & so on
-- `spec.*.spec` will mostly reflect the fields supported by Pod kind
-
-#### CStorVolumeClaim -- new custom resource
+- Default cstorVolumePolicy yamls will be present in case the user doesn't provide one.
+- App based default csstorVolumePolicies will also be present for users to pick one.
+- Whenever a user provides a cstorVolumePolicy, the empty feilds will be autofilled by webhook from the default cstorVolumePolicy yaml.
+
+#### CStorVolumeConfig -- new custom resource
 This resource is built & created by the CSI driver on a volume create request.
 This resource will be the trigger to get the desired i.e. requested cstor volume
-into the actual state in Kubernetes cluster. CStorVolumeClaim controller will 
-reconcile this claim into actual resources. CStorVolumeClaim controller will 
-operate from within the maya-apiserver pod.
+into the actual state in Kubernetes cluster. CStorVolumeConfig controller will 
+reconcile this config into actual resources. CStorVolumeConfig controller will 
+operate from within the maya-apiserver pod. Once all the below resources are 
+created successfully, the CstorVolumeConfig phase is marked as Bound.
 
-CStorVolumeClaim will be reconciled into following owned resources:
-- ConfigInjector _(Kubernetes custom resource)_
+CStorVolumeConfig will be reconciled into following owned resources:
 - CStorVolume _(Kubernetes custom resource)_
 - CStor volume target _(Kubernetes Deployment)_
 - CStor volume service _(Kubernetes Service)_
 - CStorVolumeReplica _(Kubernetes custom resource)_
 
-Following is the proposed schema for `CStorVolumeClaim`:
+Following is the proposed schema for `CStorVolumeConfig`:
 
 ```yaml
-kind: CStorVolumeClaim
+kind: CStorVolumeConfig
 metadata:
   name: <name of the volume>
   namespace: <openebs system namespace>
   labels:
   annotations:
-    openebs.io/config-class: 
     openebs.io/volume-id:
+// Spec defines a specification of a cstor volume config required
+// to provision cstor volume resources
 spec:
-  capacity:
-  claimRef:
+  // Capacity represents the actual resources of the underlying
+  // cstor volume.
+  Capacity corev1.ResourceList `json:"capacity"`
+  
+  // Policy represents the config parameters to be used for volume provisioning 
+  policy:
+    spec:
```

Comment:
can we have targetHash & replicaHash fields or annotation to identify the changes instead of iterating over all related resources and verifying the same?